# BINF6110-Assignment-1-
BINF 6110 assignment 1

Introduction:

Salmonella enterica is a major foodborne pathogen responsible for significant occurrences of illness worldwide, with some strains exhibiting different host specificities and virulence factors. Analyzing genomic variations can help shed light on the reasons for these differences and help with outbreak investigation and monitoring.

Whole-genome sequencing is a vital part of modern bacterial genomics as it enables high-resolution identification of bacterial strains, tracking of outbreaks, and identification of virulence factors and antimicrobial resistance genes. The primary goal of this study was to assemble a Salmonella enterica genome from raw reads generated by an Oxford Nanopore (ON) R10 sequencing and compare it to a reference genome in order to identify genetic variations. The workflow included quality control of raw data, de novo genome assembly, alignment to reference genome, variant calling, and final visualization of results. 
There are some challenges that may come with genome assembly from long-read sequence data. Despite the recent improvements in ON R10, achieving Q20+ accuracy (99.5%), ON reads used to exhibit higher error rates compared to short-read sequencers like Illumina (Sanderson et al., 2024). The R10 chemistry using improved basecalling models has now significantly reduced said errors, resulting in near-perfect bacterial genomes without the need for short-read polishing (Bogaerts et al., 2024). Computational demands are another challenge in processing large genomic datasets and assemblers must be able to balance accuracy with efficiency. Lastly, repetitive regions, mobile elements, and structural variations can sometimes cause issues with genome assembly, which then requires algorithms that are capable of resolving said complex genomic architectures.

This nanopore only approach simplifies workflow, eliminating the need for separate Illumina sequencing and hybrid assembly. It also allows for complete structural variation detection as long reads can span repetitive regions and resolve complex structural variants that short reads cannot. It also has a rapid turnaround, is cost effective, and has better plasmid resolution as long reads allow for complete plasmid assembly which is crucial for tracking antimicrobial resistance genes.

There are also challenges that come with alignment and variant calling. Whole-genome alignment algorithms have to be able to handle genomic rearrangements, insertions, deletions, and inversions that might occur between different strains (Saada et al., 2024). The choice of alignment method, whether they’re suffix tree-based (MUMmer), anchor-based (Minimap2), or graph-based approaches, can significantly impact the detection of structural variations and single nucleotide polymorphisms (SNPs). Despite the challenges, variant calling using long-read data has shown considerable improvement after the development of deep learning-based approaches. A recent benchmarking study by Hall demonstrated that tools like Clair3, DeepVariant, and Medaka could match or exceed the accuracy of Illumina-based variant calling in bacterial genomes. However, errors related to DNA methylation can still affect variant calling, so careful consideration of basecalling models and polishing methods are necessary (Bogaerts et al., 2025).


Methods:

In this analysis, a Nanopore-only workflow was followed using Oxford Nanopore R10 chemistry sequencing data. This approach takes advantage of the recent technological advances, making long-read only assemblies accurate without requiring hybrid approaches. 
Quality control will be performed using NanoPlot v1.46.2 (De Coster et al., 2018) to assess read length distribution, quality scores, and sequencing yield.
There are several assembler options for long-read bacterial genome assembly. Flye version 2.9.6 was chosen over others like Canu and Raven as it uses a repeat graph approach proven to perform well with bacterial genomes, and has the ability to handle long-read data efficiently while maintaining faster runtimes than Canu (Kolmogorov et al., 2019). After assembly I will use Medaka version 2.2.0 for consensus polishing as it uses neural networks trained on nanopore data to correct errors introduced in basecalling. For whole-genome alignment and variant calling I will use Minimap2 version 2.20 over MUMmer4 because while MUMmer is good for detecting structural variations, Minimap2 has improved speed and is better optimized for Nanopore reads (Li, 2018; Saada et al., 2024). Variant detection will be done also using Medaka as it is optimized for nanopore data. Recent benchmarking studies showed that it achieved F1 scores exceeding 99% for SNPs and 95% for indels with R10 data and high-accuracy basecalling (Hall et al., 2024), making it a better choice compared to BCFtools for this project. 

References:

  Bogaerts, B., Bossche, A. V. den, Verhaegen, B., Delbrassinne, L., Mattheus, W., Nouws, S., & Godfroid, M. (2024, May 4). Closing the gap: Oxford Nanopore Technologies R10 sequencing allows comparable results to illumina sequencing for SNP-based outbreak investigation of bacterial pathogens. Journal of clinical microbiology. https://doi.org/10.1128/jcm.01576-23 

  Bogaerts, B., Maex, M., Commans, F., & Goeders, N. (2025, August 22). Oxford Nanopore Technologies R10 sequencing enables accurate cgmlst-based bacterial outbreak investigation of neisseria meningitidis and salmonella enterica when accounting for methylation-related errors. Journal of clinical microbiology. https://doi.org/10.1128/jcm.00410-25 

  De Coster, W., D’Hert, S., Schultz, D. T., Cruts, M., & Van Broeckhoven, C. (2018, August 1). NanoPack: Visualizing and processing long-read sequencing data. Bioinformatics (Oxford, England). https://doi.org/10.1093/bioinformatics/bty149 

  Hall, M. B., Wick, R. R., Judd, L. M., Nguyen, A. N., Steinig, E. J., Xie, O., Davies, M., Seemann, T., Stinear, T. P., & Coin, L. (2024, October 10). Benchmarking reveals superiority of deep learning variant callers on bacterial nanopore sequence data. eLife. https://doi.org/10.7554/elife.98300 

  Kolmogorov, M., Yuan, J., Lin, Y., & Pevzner, P. A. (2019, April 1). Assembly of long, error-prone reads using repeat graphs. Nature News. https://doi.org/10.1038/s41587-019-0072-8 

  Li, H. (2018, September). Minimap2: Pairwise alignment for nucleotide sequences | bioinformatics | oxford academic. Oxford Academic. https://doi.org/10.1093/bioinformatics/bty191 

  Saada, B., Zhang, T., Siga, E., Zhang, J., & Magalhães Muniz, M. M. (2024, June 3). Whole-genome alignment: Methods, challenges, and future directions. MDPI. https://doi.org/10.3390/app14114837 

  Sanderson, N. D., Hopkins, K. M. V., Colpus, M., Parker, M., Lipworth, S., Crook, D., & Stoesser, N. (2024, May 7). Evaluation of the accuracy of bacterial genome reconstruction with Oxford nanopore R10.4.1 long-read-only sequencing. microbiologyresearch.org. https://doi.org/10.1099/mgen.0.001246 





Part 2

Methods redone/finalized:

Dataset Description
The dataset for this analysis consists of Oxford Nanopore R10 chemistry sequencing data of Salmonella enterica. The reads had an expected accuracy of Q20+ and an N50 read length ranging from 5-15 kb. The raw reads were provided in FASTQ format containing both sequence data and per-base quality scores.

Reference genome
The reference genome used for alignment and variant calling was Salmonella enterica subsp. enterica serovar Typhimurium strain LT2 (GenBank accession: NC_003197.2).

This strain was chosen because it is one of the most well-characterized S. enterica strains with a complete, high-quality genome assembly and extensive annotation, making it ideal for comparative genomics and variant detection

quality control
Quality assessment of raw Nanopore reads was performed using NanoPlot v1.46.2 (De Coster et al., 2018)


